Mage.Sets/src/mage/cards/x/XantchaSleeperAgent.java

```diff
@@ -0,0 +1,98 @@
+package mage.cards.x;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.*;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.InfoEffect;
+import mage.abilities.effects.common.LoseLifePermanentControllerEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetOpponent;
+
+
+
+/**
+ *
+ * @author jesusjbr
+ */
+
+public final class XantchaSleeperAgent extends CardImpl {
+
+
+
+    public XantchaSleeperAgent(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{B}{R}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.MINION);
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(5);
+
+        // As Xantcha, Sleeper Agent enters the battlefield, an opponent of your choice gains control of it.
+        Ability ability = new EntersBattlefieldTriggeredAbility(new XantchaSleeperAgentChangeControlEffect()) ;
+        ability.addTarget(new TargetOpponent());
+        this.addAbility(ability);
+
+        // Xantcha attacks each combat if able and can’t attack its owner or planeswalkers its owner controls.
+        this.addAbility(new AttacksEachCombatStaticAbility());
```

Comment:
doesn't it need something like to prevent attacks against owner?
        Effect effect = new CantAttackYouOrPlaneswalkerAllEffect(Duration.WhileOnBattlefield, filter);

Reply:
You are right. I forgot that part. I'll fix it later. TY!